### ISSUE TITLE
shouldBackgroundReloadRecord == true appears to ignore the cache

### ISSUE THREAD

**Issue by jerel** [emberjs/data#3484]
I'm upgrading an app to Ember 1.3.2 and Ember Data 1.3.4. To acknowledge deprecations I added:
``` js
shouldReloadAll: function() {
  return false;
},
shouldBackgroundReloadRecord: function() {
  return true;
},
```
And I'm seeing strange behavior with regards to locally cached models.

The following scenario causes the same records to be fetched from the api many times *even though all channels were preloaded by a parent model*:
``` js
  model: function() {
    return this.store.peekAll('message');
  },

  {{#each model as |message| }}
    {{message.channel.name}}
  {{/each}}
```

To display a list of messages /api/channels/1 is called ~20 times within the same runloop.

**Comment by wecc** [comment]
Thanks for reporting!

I think this should be fixed by #3468

**Comment by HeroicEric** [comment]
@jerel can you confirm this was fixed by #3468 ?

**Comment by bmac** [comment]
Closing this should have been fixed by https://github.com/emberjs/data/pull/3468.
